File: 1.py
a = [2, 3, 4, 5, 6]
def mul_all(arr):
    res = 1
    for x in arr:
        res *= x
    return res
def mul_even(arr):
    res = 1
    for x in arr:
        if x % 2 == 0:
            res *= x
    return res
print("все элементы ", mul_all(a))
print("чет элементы ", mul_even(a))

File: 1.pas
var
a: array[1..5] of integer = (2, 3, 4, 5, 6);
function mul_all: integer;
var
  i, res: integer;
begin
  res := 1;
  for i := 1 to 5 do
    res := res * a[i];
  mul_all := res;
end;
function mul_even: integer;
var
  i, res: integer;
begin
  res := 1;
  for i := 1 to 5 do
    if a[i] mod 2 = 0 then
      res := res * a[i];
  mul_even := res;
end;
begin
  writeln('Все элементы: ', mul_all);
  writeln('Четные элементы: ', mul_even);
end.